fe/fe-core/src/main/java/com/starrocks/catalog/ScalarType.java

```diff
@@ -69,7 +69,7 @@ public class ScalarType extends Type implements Cloneable {
     public static final int DEFAULT_SCALE = 0; // SQL standard
     // Longest supported VARCHAR and CHAR, chosen to match Hive.
     public static final int DEFAULT_STRING_LENGTH = 65533;
-    public static final int OLAP_MAX_VARCHAR_LENGTH = 1048576;
+    public static final int OLAP_MAX_VARCHAR_LENGTH = Config.max_varchar_length;
```

Comment:
This value is defined when compiling, but configure is defined when program running. So when the conf file is changed, it won't take affect.

Reply:
Sorta related, does the change in `be/src/storage/olap_define.h` suffer from the same problem?

Reply:
Friendly ping.

Reply:
Sorry for late reply.
How about introducing a global function which returns max length of varchar type?
will it make sense to you?

Reply:
Makes sense. I changed the code per your suggestions. Please take another look?